cis194/week4/zehua/test/ExercisesSpec.hs

```diff
@@ -0,0 +1,31 @@
+module ExercisesSpec where
+
+-- import           Control.Applicative (liftA2)
+import           Control.Monad (liftM2)
+import           Exercises
+import           Test.Hspec
+
+spec :: Spec
+spec = do
+  describe "ex1.fun1" $ do
+    it "works with empty" $ do
+      fun1' [] `shouldBe` fun1 []
+
+    it "works with non-empty" $ do
+      mapM_ (liftM2 shouldBe fun1' fun1)
```

Comment:
is this the right way to do it (via `mapM_`)?

Also I found no difference between `liftA2` and `liftM2` here.

Reply:
Yeah this is correct.

If the Applicative instance is defined using monad’s >>=, then liftA2 and liftM2 are the same.